src/transformers/generation/utils.py

```diff
@@ -1800,8 +1802,15 @@ def typeerror():
 
         return result
 
+    def contrastive_search(self, *args, **kwargs):
+        logger.warning_once(
+            "Calling `contrastive_search` directly is deprecated and will be removed in v4.41. Use `generate` or a "
+            "custom generation loop instead.",
+        )
```

Comment:
we could even leave it as is, this way custom `self._contrastive_search` are easily changeable for users? or something like that

Reply:
In the future, I don't want to have all these methods, as they are near copies of each other -- e.g. `greedy_search` and `sample` differ by <5 lines. As such, some of them will be deleted, so users won't be able to call e.g. `_greedy_search` :D